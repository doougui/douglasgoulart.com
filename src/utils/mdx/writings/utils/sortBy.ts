import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
