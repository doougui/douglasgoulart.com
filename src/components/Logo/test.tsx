import { render } from 'utils/tests';

import { Logo } from '.';

describe('<Logo />', () => {
  it('should render correctly', () => {
    const { container } = render(<Logo />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
