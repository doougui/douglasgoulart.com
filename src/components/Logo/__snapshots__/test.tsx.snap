// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render correctly 1`] = `
.c1 {
  font-family: Poppins;
  font-size: 3.2rem;
  color: #EDEDEF;
}

.c2 {
  color: #107BEB;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  class="c0"
  href="/"
>
  <div
    class="c1"
    data-testid="logo"
  >
    Douglas
    <span
      class="c2"
    >
      .
    </span>
  </div>
</a>
`;
